src/core/public/application/utils.ts

```diff
@@ -52,3 +55,55 @@ export const appendAppPath = (appBasePath: string, path: string = '') => {
     leading: false,
   });
 };
+
+export function isLegacyApp(app: App | LegacyApp): app is LegacyApp {
+  return app.legacy === true;
+}
+
+/**
+ * Converts a relative path to an absolute url.
+ * Implementation is based on a specified behavior of the browser to automatically convert
+ * a relative url to an absolute one when setting the `href` attribute of a `<a>` html element.
+ *
+ * @example
+ * ```ts
+ * // current url: `https://kibana:8000/base-path/app/my-app`
+ * relativeToAbsolute('/base-path/app/another-app') => `https://kibana:8000/base-path/app/another-app`
+ * ```
+ */
+export const relativeToAbsolute = (url: string): string => {
+  const a = document.createElement('a');
+  a.setAttribute('href', url);
+  return a.href;
```

Comment:
This is the implem that seems to be commonly used in the codebase. However I'm wondering if we shouldn't just `concat(window.location.origin, path)` instead...

The href trick has the advantage to work out of the box with the `<base>` meta tag if we choose to implement it one day.